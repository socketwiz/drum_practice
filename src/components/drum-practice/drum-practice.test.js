/* global afterEach, beforeEach, describe, expect, jest, it */

import '../../../setup';

import {shallow} from 'enzyme';
import React from 'react';
import DrumPractice from './index';
import fetchMock from 'fetch-mock';

fetchMock.get('/songs', []);

let onDragOverSpy = null;
let onDropSpy = null;

describe('DrumPractice main page', () => {
    beforeEach(() => {
        const dragEvents = {
            'onDragOver': () => {},
            'onDrop': () => {}
        };

        onDragOverSpy = jest.spyOn(dragEvents, 'onDragOver');
        onDropSpy = jest.spyOn(dragEvents, 'onDrop');
    });

    afterEach(() => {
        onDragOverSpy.mockReset();
        onDragOverSpy.mockRestore();

        onDropSpy.mockReset();
        onDropSpy.mockRestore();
    });

    it('renders DrumPractice without crashing', () => {
        const wrapper = shallow(<DrumPractice onDragOver={onDragOverSpy} onDrop={onDropSpy} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('calls onDragOver', () => {
        const wrapper = shallow(<DrumPractice onDragOver={onDragOverSpy} onDrop={onDropSpy} />);

        wrapper.find('.drop-zone').simulate('dragover');
        expect(onDragOverSpy).toHaveBeenCalled();
    });

    it('calls onDrop', () => {
        const wrapper = shallow(<DrumPractice onDragOver={onDragOverSpy} onDrop={onDropSpy} />);

        wrapper.find('.drop-zone').simulate('drop');
        expect(onDropSpy).toHaveBeenCalled();
    });
});
