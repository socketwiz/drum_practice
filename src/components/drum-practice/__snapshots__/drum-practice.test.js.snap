// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrumPractice main page renders DrumPractice without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrumPractice
    onDragOver={[Function]}
    onDrop={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="drop-zone"
          onDragOver={[Function]}
          onDrop={[Function]}
>
          <div
                    className="border"
          >
                    Drag one or more audio files to this Drop Zone ...
          </div>
</div>,
        <withRouter(Connect(YoutubeForm)) />,
        <withRouter(Connect(SongList)) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="border"
>
            Drag one or more audio files to this Drop Zone ...
</div>,
          "className": "drop-zone",
          "onDragOver": [Function],
          "onDrop": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Drag one or more audio files to this Drop Zone ...",
            "className": "border",
          },
          "ref": null,
          "rendered": "Drag one or more audio files to this Drop Zone ...",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="drop-zone"
            onDragOver={[Function]}
            onDrop={[Function]}
>
            <div
                        className="border"
            >
                        Drag one or more audio files to this Drop Zone ...
            </div>
</div>,
          <withRouter(Connect(YoutubeForm)) />,
          <withRouter(Connect(SongList)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="border"
>
              Drag one or more audio files to this Drop Zone ...
</div>,
            "className": "drop-zone",
            "onDragOver": [Function],
            "onDrop": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Drag one or more audio files to this Drop Zone ...",
              "className": "border",
            },
            "ref": null,
            "rendered": "Drag one or more audio files to this Drop Zone ...",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
